src/other/contango/ContangoWand.sol

```diff
@@ -0,0 +1,317 @@
+        debt[baseId][ilkId] = DataTypes.Debt({max: max, min: min, dec: dec, sum: 0});
+    }
+
+    /// @notice Set the debt limits for a given asset pair in the Cauldron, within bounds
+    function setDebtLimits(bytes6 baseId, bytes6 ilkId, uint96 max, uint24 min, uint8 dec) external auth {
+        // If the ilkId is a series and boundaries are not set, set them to default values
+        DataTypes.Debt memory bounds_ = debt[baseId][ilkId];
+        if (bounds_.max == 0) {
+            bounds_ = debt[baseId][yieldCauldron.series(ilkId).baseId];
+        }
+        if (bounds_.max == 0) {
+            bounds_ = yieldCauldron.debt(baseId, ilkId);
+        }
+        if (bounds_.max == 0) {
+            bounds_ = yieldCauldron.debt(baseId, yieldCauldron.series(ilkId).baseId);
+        }
+
+        uint256 paramMultiplier = 10 ** dec;
+        uint256 boundsMultiplier = 10 ** bounds_.dec;
+
+        require(max * paramMultiplier <= bounds_.max * boundsMultiplier, "Max debt out of bounds");
+        require(min * paramMultiplier >= bounds_.min * boundsMultiplier, "Min debt out of bounds");
+
+        contangoCauldron.setDebtLimits(baseId, ilkId, max, min, dec);
+    }
+
+    /// ----------------- Oracle Governance -----------------
+
+    /// @notice Set a pool as a source in the YieldSpace oracle, as long as:
+    /// - It is a pool known to the Yield Ladle
+    /// - The baseId matches the pool's baseId
+    /// - The quoteId matches the pool's seriesId
+    function setYieldSpaceOracleSource(bytes6 seriesId) external auth {
+        IPool pool_ = IPool(yieldLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        DataTypes.Series memory series_ = yieldCauldron.series(seriesId);
+        require(address(series_.fyToken) != address(0), "Series not known to the Yield Cauldron");
+        require(address(series_.fyToken) == address(pool_.fyToken()), "fyToken mismatch"); // Sanity check
+
+        yieldSpaceOracle.setSource(seriesId, series_.baseId, pool_);
+    }
+
+    /// @notice Set the YieldSpace oracle as the source for a given asset pair in the Composite oracle, provided the source is set in the YieldSpace oracle
+    function setCompositeOracleSource(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.Series memory series_ = yieldCauldron.series(ilkId);
+        if (series_.fyToken != IFYToken(address(0))) {
+            (IPool pool_,) = yieldSpaceOracle.sources(baseId, ilkId);
+            require(address(pool_) != address(0), "YieldSpace oracle not set");
+            compositeOracle.setSource(baseId, ilkId, yieldSpaceOracle);
+        } else if (yieldCauldron.assets(ilkId) != address(0)) {
+            DataTypes.SpotOracle memory spotOracle_ = yieldCauldron.lookupSpotOracle(baseId, ilkId);
+            compositeOracle.setSource(baseId, ilkId, spotOracle_.oracle);
```

Comment:
Only if it is not `IOracle(address(0))`, I think.

Reply:
the lookup method does that already